ts/main.ts

```diff
@@ -46,24 +50,45 @@ $formInputs.addEventListener('submit', (event: Event) => {
     note: $formElements.note.value,
     entryId: data.nextEntryId,
   };
-  data.nextEntryId++;
-  data.entries.unshift(obj);
-
-  $imageSubmit.setAttribute('src', '/images/placeholder-image-square.jpg');
-
-  if (!$ul) throw new Error('The $ul query failed');
-  $ul.prepend(renderEntry(obj));
   viewSwap('entries');
 
-  if (data.entries.length > 0) {
-    toggleNoEntries();
+  if (data.editing === null) {
+    data.nextEntryId++;
+    data.entries.unshift(obj);
+
+    $imageSubmit.setAttribute('src', '/images/placeholder-image-square.jpg');
+    if (!$ul) throw new Error('The $ul query failed');
+    $ul.prepend(renderEntry(obj));
+    if (data.entries.length > 0) {
+      toggleNoEntries();
+    }
+  } else if (data.editing !== null) {
+    obj.entryId = data.editing.entryId;
+    for (let i = 0; i < data.entries.length; i++) {
+      if (data.entries[i].entryId === obj.entryId) {
+        data.entries[i] = obj;
+      }
+    }
+    for (let i = 0; i < $li.length; i++) {
+      const $liDataEntryId = $li[i].getAttribute('data-entry-id');
+      if ($liDataEntryId === obj.entryId.toString()) {
+        const $newTree = renderEntry(data.entries[i]);
+        $li[i].replaceWith($newTree);
+      }
+    }
+    location.reload();
```

Comment:
We are preventing the page from refreshing by doing `event.preventDefault()`. so why would we force a reload here?